modules/platforms/dotnet/Apache.Ignite.Core/Impl/Cache/Near/NearCacheManager.cs

```diff
@@ -0,0 +1,193 @@
+
+        /// <summary>
+        /// Updates near cache from <see cref="ThreadLocalPair"/>.
+        /// </summary>
+        public void UpdateFromThreadLocal(int cacheId, int partition, AffinityTopologyVersion affinityTopologyVersion)
+        {
+            var nearCache = _nearCaches.GetOrAdd(cacheId, 
+                _ => CreateNearCache(_ignite.GetCacheConfiguration(cacheId)));
+            
+            nearCache.UpdateFromThreadLocal(partition, affinityTopologyVersion);
+        }
+
+        /// <summary>
+        /// Stops near cache.
+        /// </summary>
+        public void Stop(int cacheId)
+        {
+            INearCache cache;
+            if (_nearCaches.Remove(cacheId, out cache))
+            {
+                cache.Stop();
+            }
+        }
+
+        /// <summary>
+        /// Called when topology version changes.
+        /// </summary>
+        public void OnAffinityTopologyVersionChanged(AffinityTopologyVersion affinityTopologyVersion)
+        {
+            _affinityTopologyVersion = affinityTopologyVersion;
+        }
+        
+        /// <summary>
+        /// Creates near cache.
+        /// </summary>
+        private INearCache CreateNearCache(CacheConfiguration cacheConfiguration)
+        {
+            var nearCfg = cacheConfiguration.PlatformNearConfiguration;
+            Debug.Assert(nearCfg != null);
+            
+            Func<object> affinityTopologyVersionFunc = () => _affinityTopologyVersion;
+            var affinity = _ignite.GetAffinity(cacheConfiguration.Name);
+            var keepBinary = nearCfg.KeepBinary;
+
+            TypeResolver resolver = null;
+            Func<string, Type> resolve = n =>
+            {
+                if (n == null)
+                {
+                    return typeof(object);
+                }
+
+                if (resolver == null)
+                {
+                    resolver = new TypeResolver();
+                }
+
+                var resolved = resolver.ResolveType(n);
+
+                if (resolved == null)
+                {
+                    throw new InvalidOperationException(string.Format("Failed to resolve type: '{0}'", n));
+                }
+
+                return resolved;
+            };
+
+            var keyType = resolve(nearCfg.KeyTypeName);
+            var valType = resolve(nearCfg.ValueTypeName);
+
+            var cacheType = typeof(NearCache<,>).MakeGenericType(keyType, valType);
+            var nearCache = Activator.CreateInstance(
+                cacheType, 
+                affinityTopologyVersionFunc, 
+                affinity,
+                keepBinary);
+            
+            return (INearCache) nearCache;
+        }
+        
+        /// <summary>
+        /// Handles client disconnect.
+        /// </summary>
+        private void OnClientDisconnected(object sender, EventArgs e)
+        {
+            foreach (var cache in _nearCaches)
+            {
+                cache.Value.Clear();
```

Comment:
FYI: Could hit performance, customers could use a thick client for health checking instead of a thin one

Reply:
I think that compared to the whole "thick client connect/disconnect" procedure, this call is very cheap.